app/users/users.go

```diff
@@ -30,7 +30,13 @@ type savedAccFields struct {
 
 // Retriever is an interface for user retrieval by internal-apis auth token
 type Retriever interface {
-	Retrieve(token string) (*models.User, error)
+	Retrieve(query Query) (*models.User, error)
+}
+
+// Query contains queried user details and optional metadata about the request
+type Query struct {
+	Token        string
+	MetaRemoteIP string
```

Comment:
why "meta"?

Reply:
It's metadata for the query, does not affect the query itself.